### ISSUE TITLE
Force ID-based interface functions to use names in error messages

### ISSUE THREAD

**Issue by fsimonis** [precice/precice#688]
**Please describe the problem you are trying to solve?**
preCICE error messages during user input sanitization can be quite confusing.
Error messages handling mesh or data IDs should always print the name and the ID as long as they exist.

**Describe the solution you propose.**
Revisit all ID-based functions in the SolverInterface and add the mesh/data name to error messages.

**Describe alternatives you've considered**
_None_

**Additional context**
@maierbn already ran into a non-sensical message in #680

**Comment by fsimonis** [comment]
It may also be very helpful to extend the error message by possible choices.

```
ERROR: This participant does not know a Mesh named "ToFu"!
Maybe you meant one of the following: "Soja", "Nutils", "Bacon"?
```

**Comment by fsimonis** [comment]
Solved by #721
